documentation: Add Reference to the remote read API (#7266)
prometheus/prometheus: docs/querying/api.md

*New in v2.15*

> ## Remote Read API

Reviewer: ```suggestion
## Remote read API
```

> This API provides data read functionality from prometheus. This interface expects [snappy](https://github.com/google/snappy) compression.

Reviewer 2: This should follow the style of the rest of the docs.

Reviewer: Can you clarify what you mean by this?

Author: @brian-brazil Is there a particular style guide?

Reviewer 2: There isn't (and it'd be too granular to cover this), so look at the rest of the page for how we word things.

> This API provides data read functionality from prometheus. This interface expects [snappy](https://github.com/google/snappy) compression.

Reviewer: ```suggestion
This API provides data read functionality from Prometheus. This interface expects [snappy](https://github.com/google/snappy) compression.
```

Author: Updated locallly (will appear in next patch)

The API definition is located [here](https://github.com/prometheus/prometheus/blob/master/prompb/remote.proto).

> Request are made to the following endpoint.

Reviewer 3: Can we talk a little bit how negotiation works of sampled vs streamed ?

```
/api/v1/read
```

### Samples

> This will return a message that includes a list of raw samples.

Reviewer: ```suggestion
This returns a message that includes a list of raw samples.
```

Reviewer: Does it return a message that includes a list, or does it return a list of raw samples?

> ### Streamed Chunks

Reviewer: ```suggestion
### Streamed chunks
```

> These streamed chunks utilize an XOR algorithm inspired by the [Gorilla](http://www.vldb.org/pvldb/vol8/p1816-teller.pdf)

Reviewer 2: This is a little much detail, the relevant point is they're the internal implementation of the chunks.
